Tasks/RunDistributedTests/RunDistributedTests.ps1

```diff
@@ -58,6 +147,23 @@ $checkTestAgentCompatScriptLocationMemberExists  = CmdletHasMember "CheckTestAge
 $checkCustomSlicingEnabledMemberExists  = CmdletHasMember "CustomSlicingEnabled"
 $taskContextMemberExists  = CmdletHasMember "TaskContext"
 
+if($overrideRunParams -and $runSettingsFile -and (Test-Path $runSettingsFile))
+{
+    if (([string]::Compare([io.path]::GetExtension($runSettingsFile), ".testsettings", $True) -eq 0)) {
+        $xml = [xml](Get-Content $runSettingsFile)
+        $hasOverridenProperties = OverrideTestSettingProperties -overridingParameters $overrideRunParams -testSettings $xml
```

Comment:
do check for empty and invalid xml

Reply:
added the check

---
In reply to: [137521290](https://github.com/Microsoft/vsts-tasks/pull/5246#discussion_r137521290) [](ancestors = 137521290)